src/cmd/flux-jobs.py

```diff
@@ -168,23 +168,36 @@ def fetch_jobs_flux(args):
                 print("invalid user specified", file=sys.stderr)
                 sys.exit(1)
 
-    flags = 0
+    states = 0
     for state in args.states.split(","):
-        if state.lower() == "pending":
-            flags |= flux.constants.FLUX_JOB_LIST_PENDING
+        # Note
+        # pending = depend & sched
+        # running = run & cleanup
+        if state.lower() == "depend":
+            states |= flux.constants.FLUX_JOB_DEPEND
+        elif state.lower() == "sched":
+            states |= flux.constants.FLUX_JOB_SCHED
+        elif state.lower() == "pending":
+            states |= flux.constants.FLUX_JOB_PENDING
+        elif state.lower() == "run":
+            states |= flux.constants.FLUX_JOB_RUN
+        elif state.lower() == "cleanup":
+            states |= flux.constants.FLUX_JOB_CLEANUP
```

Comment:
Feel free to take or leave this suggestion, but I think you can cut the number of lines in half by building a dictionary mapping strings to state constants and then just doing a lookup (the dict could also be built outside the func).  E.g.:
```python
state_const_dict = {
    "depend" : flux.constants.FLUX_JOB_DEPEND,
    ...
    "active" : flux.constants.FLUX_JOB_ACTIVE,
}

for state in args.states.split(","):
  try:
    states |= state_const_dict[state.lower()]
  except KeyError:
    print("Invalid state")
```

Reply:
Good idea,  i'll try something out

Reply:
Actually should we add that to the Python `job.job_list()` function itself to make it more user friendly? i.e. instead of passing in a state mask you can just pass a `states` list directly.

Reply:
that's a good idea, only thought is that `flags` is also a bitmask, for now atleast.  And if #2623 is resolved, perhaps the code will change to a direct call to the function.

I'd say lets put this on maybe list for later?

Reply:
Yeah, I'm fine with putting it on the list for later, though my opinion is as much as possible we wrap the raw C calls in the Python bindings to make them more pythonic.

I'm still confused as to what `flags` is used for. Currently valid flags is 0?

Reply:
Yeah, `flags` is not used for anything now, so its there as a placeholder just in case we need it for something in the future.

Reply:
I argue we should remove it. I can't think of anything we'd use it for, realistically we're going to have to change the API in the future anyway if we want to add real filter/query capability to the interface, e.g. filtering by job/command name, etc. Even for something like sorting, you'd need to send what key you want to sort on... Perhaps we have something in mind for flags I'm just not thinking of?

Reply:
I can't really think of anything.   You are right, we're going to add new parameters anyways for some of the other issues we've been talking about.  I'll axe it and change the commit to say "rename flags to states".